### PR TITLE
Feature/nakijin 1828 mapping with giata data

### PR DESCRIPTION
**Pull request by hasmikharutyunyan**

### CODE REVIEW

File: HappyTravel.Nakijin.Data/Models/GiataIdSupplierCode.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+namespace HappyTravel.Nakijin.Data.Models;
+
+public class GiataIdSupplierCode
+{
+    public int GiataId { get; init; }
+    public string GiataSupplier { get; init; }
```
Comment (oleg-n-k) [inline HappyTravel.Nakijin.Data/Models/GiataIdSupplierCode.cs:6]: And just supplier?

Comment (hasmikharutyunyan) [reply HappyTravel.Nakijin.Data/Models/GiataIdSupplierCode.cs:6]: This is supplier name by Giata format

File: HappyTravel.Nakijin.Updater/Workers/Giata/MergedAccommodationProcessor.cs [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    public MergedAccommodationProcessor(NakijinContext context)
+    {
+        _context = context;
+    }
+
+
+    public async Task ProcessMergedAccommodationsUncertainMatches(List<KeyValuePair<int, List<int>>> mergedAccommodations, CancellationToken cancellationToken)
+    {
+        var existingUncertainMatches =
+            await GetMergedAccommodationsUncertainMatches(mergedAccommodations.Select(ma => ma.Key)
```
[the remark this answers is not in the document]
Comment (hasmikharutyunyan) [reply HappyTravel.Nakijin.Updater/Workers/Giata/MergedAccommodationProcessor.cs:23]: you mean variable for `mergedAccommodations.Select(ma => ma.Key)`

File: HappyTravel.Nakijin.Updater/Workers/Giata/MergedAccommodationProcessor.cs [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                .Union(mergedAccommodations.SelectMany(m => m.Value)).ToList(), cancellationToken);
+        var uncertainMatchesToDeactivate = new List<AccommodationUncertainMatches>();
+        var nowDate = DateTimeOffset.UtcNow;
+
+        var mergedPairs = new List<KeyValuePair<int, int>>();
+
+        foreach (var mergedData in mergedAccommodations)
+            mergedPairs.AddRange(mergedData.Value.Select(v => new KeyValuePair<int, int>(mergedData.Key, v)));
```
Comment (oleg-n-k) [inline HappyTravel.Nakijin.Updater/Workers/Giata/MergedAccommodationProcessor.cs:31]: I think it's better to use a separate variable

File: HappyTravel.Nakijin.Updater/Workers/Giata/MergedAccommodationProcessor.cs [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        _context.UpdateRange(uncertainMatchesToDeactivate);
+
+        await _context.SaveChangesAsync(cancellationToken);
+    }
+
+
+    public async Task ProcessMergedAccommodationsHtAccommodationMappings(List<KeyValuePair<int, List<int>>> mergedAccommodations,
+        CancellationToken cancellationToken)
+    {
+        var existingMappings = await GetMergedAccommodationsMappings(mergedAccommodations.Select(ma => ma.Key)
```
Comment (oleg-n-k) [inline HappyTravel.Nakijin.Updater/Workers/Giata/MergedAccommodationProcessor.cs:56]: It's better to use a variable here

File: HappyTravel.Nakijin.Updater/Workers/Giata/MergedAccommodationProcessor.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (oleg-n-k) [reply HappyTravel.Nakijin.Updater/Workers/Giata/MergedAccommodationProcessor.cs:23]: mergedAccommodations.Select(ma => ma.Key).Union(mergedAccommodations.SelectMany(m => m.Value)).ToList()

File: HappyTravel.Nakijin.Data/Models/GiataIdSupplierCode.cs [hunk 5]
```diff
@@ -0,0 +1,8 @@
+namespace HappyTravel.Nakijin.Data.Models;
+
+public class GiataIdSupplierCode
+{
+    public int GiataId { get; init; }
```
[the remark this answers is not in the document]
Comment (hasmikharutyunyan) [reply HappyTravel.Nakijin.Data/Models/GiataIdSupplierCode.cs:7]: Added Comment

File: HappyTravel.Nakijin.Data/Models/GiataIdSupplierCode.cs (hunk 1 above)
Comment (hasmikharutyunyan) [reply HappyTravel.Nakijin.Data/Models/GiataIdSupplierCode.cs:10]: Added Comment

File: HappyTravel.Nakijin.Updater/Workers/Giata/MergedAccommodationProcessor.cs (hunk 3 above)
Comment (hasmikharutyunyan) [reply HappyTravel.Nakijin.Updater/Workers/Giata/MergedAccommodationProcessor.cs:35]: I think here no need to use separate variable, as expression is too short

**Review by oleg-n-k — APPROVED** [review]

**Pull request merged**